t/plugin/proxy-mirror.t

```diff
@@ -445,3 +445,139 @@ GET /t
+                            },
+                            "type": "roundrobin"
+                        },
+                        "uri": "/hello"
+                   }]]
+                   )
+
+               if code >= 300 then
+                   ngx.status = code
+               end
+               ngx.print(body)
+           }
+       }
+--- request
+GET /t
+--- error_code: 400
+--- response_body
+{"error_msg":"failed to check the configuration of plugin proxy-mirror err: property \"sample_ratio\" validation failed: expected 10 to be smaller than 1"}
+--- no_error_log
+[error]
+
+
+
+=== TEST 14: set mirror requests sample_ratio to 1
+--- config
+       location /t {
+           content_by_lua_block {
+               local t = require("lib.test_admin").test
+               local code, body = t('/apisix/admin/routes/1',
+                    ngx.HTTP_PUT,
+                    [[{
+                        "plugins": {
+                            "proxy-mirror": {
+                               "host": "http://127.0.0.1:1986",
+                               "sample_ratio": 1
+                            }
+                        },
+                        "upstream": {
+                            "nodes": {
+                                "127.0.0.1:1980": 1
+                            },
+                            "type": "roundrobin"
+                        },
+                        "uri": "/hello"
+                   }]]
+                   )
+
+               if code >= 300 then
+                   ngx.status = code
+               end
+               ngx.say(body)
+           }
+       }
+--- request
+GET /t
+--- error_code: 200
+--- response_body
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 15: hit route with sample_ratio 1
+--- request
+GET /hello?sample_ratio=1
+--- error_code: 200
+--- response_body
+hello world
+--- error_log_like eval
+qr/uri: \/hello\?sample_ratio=1/
+
+
+
+=== TEST 16: set mirror requests sample_ratio to 0.5
+--- config
+       location /t {
+           content_by_lua_block {
+               local t = require("lib.test_admin").test
+               local code, body = t('/apisix/admin/routes/1',
+                    ngx.HTTP_PUT,
+                    [[{
+                        "plugins": {
+                            "proxy-mirror": {
+                               "host": "http://127.0.0.1:1986",
+                               "sample_ratio": 0.5
+                            }
+                        },
+                        "upstream": {
+                            "nodes": {
+                                "127.0.0.1:1980": 1
+                            },
+                            "type": "roundrobin"
+                        },
+                        "uri": "/hello"
+                   }]]
+                   )
+
+               if code == 200 then
+                   for i = 1, 200 do
+                       t('/hello?sample_ratio=0.5', ngx.HTTP_GET)
```

Comment:
Could you give some tips~ @spacewander 
Current the hit count is checked via: `qr/(uri: \/hello\?sample_ratio=0\.5){75,125}/`

Reply:
We can add a `/inc` API, it will return an increasing number each call. Like,
```
GET /inc
1
GET /inc
2
GET /inc
3
```

So by hitting the API we can know how many requests are mirrored.

Reply:
Thanks firstly~
According to the document, seems the mirror address only support host~
@spacewander 

https://github.com/apache/apisix/blob/master/docs/en/latest/plugins/proxy-mirror.md#attributes

![image](https://user-images.githubusercontent.com/75366457/132455664-7d7f9693-4112-4e37-afc3-30ffd7c1cccf.png)

Reply:
Yes, both the client and mirror hit the /inc API, so the number of request from mirror is total - client.

Reply:
okay~ will write a new backend API~